PyInstaller/building/datastruct.py

```diff
@@ -173,6 +178,17 @@ def __postinit__(self):
             else:
                 # create a dict for easier access
                 data = dict(zip((g[0] for g in self._GUTS), data))
+
+                # this ::HACK:: converts winmanifest from a string
+                if is_win and data.get('manifest'):
+                    if not data['manifest'].startswith(winmanifest_marker):
```

Comment:
What was in there prior to you change? I'd just pass the read value to winmanifest.ManifestFromXML() and if it fails, the data was not valid :-)

BTW: I just discovered that `EXE.manifest` is already build in "EXE.__init__`. I think it would be more appropriate to only add the manifest's filename to `self.toc` and generate the manifest itself later in `EXE.assemble()` 
In `create_manifest` you could even use the prior value (which you red vis _GUTS) and save reading it again.
Do you want to try this, too? This *may* save your change about comparing the file alltogether. (*May*, since I do not really know if the manifest depends on other parameters except of those passed to `winmanifest.create_manifest()`.)